src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -628,6 +648,55 @@ private void processLogsAndUpload() throws InterruptedException {
         }
     }
 
+    /**
+     * calculate the hash for the active file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, or incorrect setting for DEFAULT_LINES_FOR_DIGEST_NUM, the hash is null.
+     * @param activeFile
+     * @return the nullable calculated hash for active file
+     */
+    private Optional<String> calculateHashForActiveFile(File activeFile) {
+        Optional activeFileHash = Optional.ofNullable(null);
+        // String activeFileHash = null;
+        try (SeekableByteChannel chan = Files.newByteChannel(activeFile.toPath(), StandardOpenOption.READ);
+             PositionTrackingBufferedReader r =
+                new PositionTrackingBufferedReader(new InputStreamReader(Channels.newInputStream(chan),
+                        StandardCharsets.UTF_8))) {
+            r.setInitialPosition(0);
+            chan.position(0);
+            StringBuilder data = new StringBuilder(r.readLine());
+
+            // if the lines for digest is less than 1 line, skip this file and generate error.
+            if (DEFAULT_LINES_FOR_DIGEST_NUM < 1) {
```

Comment:
You're checking `1 < 1`, we always know that is false, this can not ever be true.

Reply:
This check is written for a future case: if the customer is allowed to customize the `DEFAULT_LINES_FOR_DIGEST_NUM`, then any non-positive integers will be rejected.

Reply:
If you do add that, this isn't where that logic would go. This is premature, so I think you should remove it.

Reply:
+1